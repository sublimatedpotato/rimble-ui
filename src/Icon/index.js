import React from 'react';
import PropTypes from 'prop-types';
import StyledIcon from 'rmdi/lib/Icon';

const Icon = props => <StyledIcon {...props} />;

Icon.defaultProps = {
  display: 'block',
  size: '24px',
};
Icon.propTypes = {
  /**
   * Sets which icon to display
   */
  name: PropTypes.string.isRequired,
  /**
   * Sets the element display property
   */
  display: PropTypes.string,
  /**
   * Sets the width and height in pixels
   */
  size: PropTypes.string,
};

Icon.displayName = 'Icon';

export default Icon;
