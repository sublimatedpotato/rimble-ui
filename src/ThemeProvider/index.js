import React from 'react';
import PropTypes from 'prop-types';

import styled, {
  createGlobalStyle,
  ThemeProvider as StyledThemeProvider,
} from 'styled-components';

import { normalize, transparentize } from 'polished';

import theme from '../theme';

export const Base = styled.div`
  & {
    font-size: 1em;
    font-family: ${props => props.theme.fonts.sansSerif};
    color: ${props => props.theme.colors.copyColor};
  }
`;

const GlobalStyle = createGlobalStyle`
  ${normalize()}
  html {
    box-sizing: border-box;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  ::selection {
    background: ${props => transparentize(0.7, props.theme.colors.primary)};
  }
`;

const ThemeProvider = ({ ...props }) => {
  return (
    <StyledThemeProvider theme={theme} {...props}>
      <React.Fragment>
        <GlobalStyle />
        <Base {...props} />
      </React.Fragment>
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {};

export default ThemeProvider;
