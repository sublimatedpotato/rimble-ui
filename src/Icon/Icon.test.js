import Icon from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Icon component sanity', () => {
  it('has name', () => {
    expect(Icon.displayName).toBe('Icon');
  });

  it('matches default snapshot', () => {
    const component = render(<Icon name={'Info'} />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Icon name={'Info'} />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
