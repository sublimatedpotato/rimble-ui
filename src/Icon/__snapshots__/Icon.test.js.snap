// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component sanity matches default snapshot 1`] = `
<svg
  class=""
  display="block"
  fill="currentcolor"
  height="24px"
  viewBox="0 0 24 24"
  width="24px"
>
  <path
    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
  />
</svg>
`;

exports[`Icon component sanity matches themed snapshot 1`] = `
.c0 {
  font-size: 1em;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  color: #3F3D4B;
}

<div
  class="c0"
>
  <svg
    class=""
    display="block"
    fill="currentcolor"
    height="24px"
    viewBox="0 0 24 24"
    width="24px"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
    />
  </svg>
</div>
`;
